src/hooks/useKeyboard.ts

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import {Keyboard} from 'react-native';
+
+export function useKeyboard() {
+  const [isVisible, setIsVisible] = React.useState(false);
+
+  React.useEffect(() => {
+    const keyboardDidShow = Keyboard.addListener('keyboardDidShow', () => {
+      setIsVisible(true);
+    });
+
+    const keyboardDidHide = Keyboard.addListener('keyboardDidHide', () => {
+      setIsVisible(false);
+    });
+
+    return () => {
+      keyboardDidShow.remove();
+      keyboardDidHide.remove();
+    };
+  }, []);
+
+  return isVisible;
```

Comment:
To make it easier to understand what this hook does we could return `{ isVisible }`, or call the hook `useKeyboardStatus` and return `'visible' | 'hidden'`